modules/affile/file-opener.c

```diff
@@ -106,7 +106,7 @@ _obtain_capabilities(FileOpener *self, const gchar *name, cap_t *act_caps)
       g_process_enable_cap("cap_dac_override");
     }
 
-  if (self->options->create_dirs &&
+  if (self->options->create_dirs == TRUE &&
```

Comment:
`create_dirs` should be checked only after calling `file_opener_options_init()`, which is responsible for getting rid of all the temporary `-1`s. I wouldn't check explicitly for `TRUE` here.

Reply:
If we don't have explicit checking for `TRUE` value, then we don't have any safety-mechanism.
> After initializing *Options objects, all of these values should be replaced with the inherited value.

Yes, but this was missing and nothing protected us from it.
I know this is a general issue.

Now, `file_opener_options_init` is initializing `create_dirs`, but I would think it more safe if there would be checking/assertion where `create_dirs` is used. Do you think a type-check(TRUE or FALSE, but nothing else) assertion in `_obtain_capabilities` would be a  good idea?
After writing it down, I don't think. :)
A function should not check for type-safety ....

I'll just remove the explicit check then. What do you think? A simple ":+1" is a good sign.

Reply:
I've dropped the patch.

Reply:
>  Do you think a type-check(TRUE or FALSE, but nothing else) assertion in _obtain_capabilities would be a good idea?

It doesn't add anything useful in my opinion, unless you do it for all of our options (or some kind of abstraction that validates all options against `-1` automatically in `init()`).